### PR TITLE
Clean option for discarded tabs

### LINKED ISSUE #720: [FEATURE REQUEST]  clean cookies for discarded tabs
**Is your feature request related to a problem? Please describe.**

I use extensions that discard/unload tabs from memory (AutoTabDiscard to name one).
When a tab is discarded, it's unloaded from memory, and reloaded later.
I expect that cookies are removed for discarded tabs, as in my workflow it's the same if I clos the tab and reopen it later.

**Describe the solution you'd like**

Launch the clean timer when the tab is discarded.

**Describe alternatives you've considered**

I tried to build a solution by myself, but it's the first time I read TypeScript and do WebExtension developpement, and I don't understand how the cookies are deleted.

**Additional context**

See attached draft. #721

### PR DESCRIPTION
**Pull request by Grumph**
Not working yet, as I don't understand how the cookies are actually deleted.

Fix #720

### CODE REVIEW

File: src/services/TabEvents.ts [hunk 1]
```diff
@@ -46,6 +46,18 @@ export default class TabEvents extends StoreUser {
...
   }
 
+  public static onTabDiscarded(
+    tabId: number,
+    changeInfo: any,
+    tab: browser.tabs.Tab,
+  ) {
+    if (tab.discarded) {
+      // Do the magic !
+      TabEvents.cleanFromFromTabEvents();
```
Comment (Grumph) [inline src/services/TabEvents.ts]: I need help here.

**Comment by Grumph** [comment]
I've found where the problem was: I had to ignore discarded tabs from the list of opened tabs.

Now this patch is working as I expect it.

It just needs some code cleaning I guess, it would be awesome if somebody can help me with that.

**Comment by kennethtran93** [comment]
Thank you for your contribution.

While the code you have drafted up may work in your case, it does need cleaning up, and will need to be adapted to the code that I will be pushing soon via #713.  

Here's some notes for future contributions:
- You've also missed putting in the translation string in `extensions/_locales/en/messages.json`
- Additional test cases to supplement the changes should be made as well.

Other than that it looks okay.

Seeing as the changes aren't too much, I'll take in the necessary code needed to make it work and push it into #713 so it gets released altogether.

**Comment by kennethtran93** [comment]
Originally I was going to just insert your edits to what I'm pushing as #713, but then you wouldn't be part of the contributed team.

At this time, please wait until I have pushed #713 to the 3.X.X Branch before further changes are made.

Thanks!

**Comment by Grumph** [comment]
I added the tests and locale to learn by doing and maybe ease your work.
Do as you prefer, I don't mind to be or not in the contributors list. What's important to me is the feature.

Thanks for this extension by the way :)

**Comment by kennethtran93** [comment]
Fair enough - if you so desire then I'll start working on patching it up myself based on your original patch.

...but it will be later on this week.

**Comment by kennethtran93** [comment]
...later on this week being now I guess.

Your contributions here have been taken through to #753.

**Pull request closed without merging**